### ISSUE TITLE
Support search for specific header

### ISSUE THREAD

**Issue by exander77** [ProtonMail/proton-mail#57]
Search for specific header presence and value should be supported in WebUI, it would be best to do it generally, but the most important are:
Subject, From, To (already supported), Cc, Bcc, Message-Id, List-Unsubscribe, X-Pm-Origin etc.

**Comment by bartbutler** [comment]
message-id is already supported by the server (though probably not in the UI), so that would be possible to add in the advanced search. Generic header search is possible of course but unlikely to happen due to relatively low benefit vs. index size I think.

**Comment by exander77** [comment]
@bartbutler It would be nice to support at least what is already indexed for other purposes and support it through search operators:
https://github.com/ProtonMail/proton-mail/issues/55

I also made some initial pull request: https://github.com/ProtonMail/proton-mail/pull/56

**Comment by bartbutler** [comment]
I agree. Using 'in:' to filter by multiple labels/folders should also be supported by the back end.
